js/src/TooltipContainer.js

```diff
@@ -20,53 +23,81 @@ TooltipContainer.prototype = {
   show: show,
   hide: hide,
   delay_hide: delay_hide,
-  cancel_hide_tooltip: cancel_hide_tooltip,
+  cancelHideTooltip: cancelHideTooltip
 }
 module.exports = TooltipContainer
 
 // definitions
-function init (selection, tooltip_component, zoom_container) {
+function init (selection, TooltipComponent, zoomContainer) {
   this.div = selection.append('div').attr('id', 'tooltip-container')
-  this.div.on('mouseover', this.cancel_hide_tooltip.bind(this))
-  this.div.on('mouseleave', this.hide.bind(this))
+  this.TooltipComponent = TooltipComponent
+  this.tooltipRef = null
+
+  this.zoomContainer = zoomContainer
+  this.setup_zoom_callbacks(zoomContainer)
+
+  // Create callback manager
+  this.callback_manager = CallbackManager()
 
-  this.setup_zoom_callbacks(zoom_container)
+  this.div.on('mouseover', this.cancelHideTooltip.bind(this))
+  this.div.on('mouseleave', this.hide.bind(this))
 
-  // keep a reference to tinier tooltip
-  this.tooltip_component = tooltip_component
-  // if they pass in a function, then use that
-  this.tooltip_function = (_.isFunction(tooltip_component) ?
-                           function (state) { tooltip_component({ state: state, el: this.div.node() })} :
-                           null)
-  // if they pass in a tinier component, use that
-  this.tinier_tooltip = (tinier.checkType(tinier.COMPONENT, tooltip_component) ?
-                         tinier.run(tooltip_component, this.div.node()) :
-                         null)
+  // Attach a function to get size of tooltip
```

Comment:
would be good to provide a little more detail in this comment, since this is a weird thing we have to do

Reply:
/**
   * Attach a function to get size of tooltip. This is done so that the show
   * function knows how much to displace rendering by. Works by using the
   * tooltip's get_size function.
  */

Is this good? I forgot how exactly this worked.